src/towncrier/test/test_settings.py

```diff
@@ -35,6 +37,53 @@ def test_base(self):
         self.assertEqual(config.underlines, ["=", "-", "~"])
         self.assertEqual(config.orphan_prefix, "~")
 
+    def test_markdown(self):
+        """
+        If the filename references an .md file and the builtin template doesn't have an
+        extension, add .md rather than .rst.
+        """
+        temp = self.mktemp()
+        os.makedirs(temp)
+
+        with open(os.path.join(temp, "pyproject.toml"), "w") as f:
+            f.write(
+                """[tool.towncrier]
+package = "foobar"
+filename = "NEWS.md"
+"""
+            )
+
+        config = load_config(temp)
+        self.assertEqual(config.filename, "NEWS.md")
+        expected_template = pkg_resources.resource_filename(
+            "towncrier", "templates/default.md"
+        )
+        self.assertEqual(config.template, expected_template)
```

Comment:
let's separate act and assert.

```suggestion
        )

        self.assertEqual(config.template, expected_template)
```